packages/enhanced/src/lib/container/runtime/RemoteEntryPlugin.ts

```diff
@@ -0,0 +1,32 @@
+import type { Compiler, WebpackPluginInstance } from 'webpack';
+import pBtoa from 'btoa';
+
+export class RemoteEntryPlugin implements WebpackPluginInstance {
+  readonly name = 'VmokRemoteEntryPlugin';
+  private _name: string;
+  private _getPublicPath: string;
+
+  constructor(name: string, getPublicPath: string) {
+    this._name = name;
+    this._getPublicPath = getPublicPath;
+  }
+
+  apply(compiler: Compiler): void {
+    let code;
+    if (!this._getPublicPath.startsWith('function')) {
+      code = `${
+        compiler.webpack.RuntimeGlobals.publicPath
+      } = new Function(${JSON.stringify(this._getPublicPath)})()`;
+    } else {
+      code = `(${this._getPublicPath})(${compiler.webpack.RuntimeGlobals.publicPath})`;
```

Comment:
Can we not pass compiler.webpack.RuntimeGlobals.publicPath to getPublicPath ?  Because in current snapshot design, getPublicPath and publicPath are mutually exclusive, so in getResourceUrl function , we can not provide the original publicPath which may cause some error if user depend on the parameter. 

![image](https://github.com/module-federation/core/assets/41466093/4fc09811-c464-41f3-8592-abaccdaf232c)

Reply:
And the data transmitted here and the data transmitted above